beacon-chain/rpc/prysm/validator/validator_performance.go

```diff
@@ -0,0 +1,78 @@
+package validator
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/rpc/core"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v4/network"
+	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+)
+
+type ValidatorPerformanceRequest struct {
+	PublicKeys [][]byte                    `json:"public_keys,omitempty"`
+	Indices    []primitives.ValidatorIndex `json:"indices,omitempty"`
+}
+
+type ValidatorPerformanceResponse struct {
+	PublicKeys                    [][]byte `json:"public_keys,omitempty"`
+	CorrectlyVotedSource          []bool   `json:"correctly_voted_source,omitempty"`
+	CorrectlyVotedTarget          []bool   `json:"correctly_voted_target,omitempty"`
+	CorrectlyVotedHead            []bool   `json:"correctly_voted_head,omitempty"`
+	CurrentEffectiveBalances      []uint64 `json:"current_effective_balances,omitempty"`
+	BalancesBeforeEpochTransition []uint64 `json:"balances_before_epoch_transition,omitempty"`
+	BalancesAfterEpochTransition  []uint64 `json:"balances_after_epoch_transition,omitempty"`
+	MissingValidators             [][]byte `json:"missing_validators,omitempty"`
+	InactivityScores              []uint64 `json:"inactivity_scores,omitempty"`
+}
+
+// GetValidatorPerformance is an HTTP handler for GetValidatorPerformance.
+func (vs *Server) GetValidatorPerformance(w http.ResponseWriter, r *http.Request) {
+	if vs.SyncChecker.Syncing() {
+		handleHTTPError(w, "Syncing", http.StatusInternalServerError)
```

Comment:
```suggestion
		handleHTTPError(w, "Syncing", http.StatusInternalServerError)
```
Syncing should return 503 status code